### PR TITLE
Add support for OTLP over HTTP/protobuf

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Adds support for ingesting OTLP requests over HTTP with protobuf data. Changs in this PR:

- Move the existing gRPC export handler and helpers to to otlp_trace.go and tests into otlp_trace_test.go
- Add HTTP handler `postOTLP` that retrieves honeycomb headers, decodes the request body into a OTLP ExportTraceRequest and hands to internal function to process request
- Update OTLP grpc Export handler to retrieve honeycomb headers from metadata and hand request to new internal function to process
- Adds tests to verify rejection behaviour for non-protobuf based requests to postOTLP and test to pass request to postOTLP handler endpoint

### CODE REVIEW

**Comment by MikeGoldsmith** [comment]
Most recent commits was to add support for gzip/zstd encoding as part of the parsing process.

**Review by vreynolds — CHANGES_REQUESTED** [review]
Route requires an extra `/1/` path

File: route/route.go [hunk 1]
```diff
@@ -161,6 +156,9 @@ func (r *Router) LnS(incomingOrPeer string) {
 	authedMuxxer.HandleFunc("/events/{datasetName}", r.event).Name("event")
 	authedMuxxer.HandleFunc("/batch/{datasetName}", r.batch).Name("batch")
 
+	// OTLP trace
+	authedMuxxer.HandleFunc("/v1/traces", r.postOTLP).Name("otlp")
```
[the remark this answers is not in the document]
Comment (MikeGoldsmith) [reply route/route.go:160]: Hmm, I thought I tested this but must have been confused with testing Hound. Thanks for catching @vreynolds 👍🏻

[the remark this answers is not in the document]
Comment (vreynolds) [reply route/route.go:160]: Do you actually see OTLP data posted to Honeycomb? I'm seeing the same logs, but as I read them, it's telling us `proxying request for /v1/traces` which got [here](https://github.com/honeycombio/refinery/blob/8c28b313627e3f0ec0635a3fc827273d4d7f3b52/route/proxy.go#L14), so I don't think the router actually used the OTLP handler, right?

In the client, I'm seeing `OpenTelemetry error: Unable to export 10 spans`

[the remark this answers is not in the document]
Comment (vreynolds) [reply route/route.go:160]: If I change the client OTLP endpoint to `localhost/1` Refinery logs `handled otlp request tSqX0iR9 [::1]:63470 POST /1/v1/traces 0.794175 200` and spans come through

[the remark this answers is not in the document]
Comment (MikeGoldsmith) [reply route/route.go:160]: Yeah, I got data through to my local Shepherd that I can query on in the UI.

otlp-http-refinery

[the remark this answers is not in the document]
Comment (MikeGoldsmith) [reply route/route.go:160]: I was using the [ruby-otlp](https://github.com/honeycombio/examples/tree/main/ruby-otlp) example and set the following env vars:
```
export OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:18080
export OTEL_EXPORTER_OTLP_HEADERS=x-honeycomb-team=a012e0a06d90af50a61a5ce4e9e19737,x-honeycomb-dataset=ruby-otlp
export SERVICE_NAME=ruby-otlp
``

[the remark this answers is not in the document]
Comment (vreynolds) [reply route/route.go:160]: For posterity: we paired on zoom, and it looks like the crux is using local shepherd vs dogfood. It still looks like requests aren't handled properly by refinery, but when paired with local shepherd, proxied (unhandled) requests still get processed, whereas when paired with dogfood, they get rejected.

[the remark this answers is not in the document]
Comment (MikeGoldsmith) [reply route/route.go]: I've added a new `otlpmuxxer` that handles OTLP requests with the `/v1/` prefix in [49732c1](https://github.com/honeycombio/refinery/pull/279/commits/49732c13447b5308f3835019f8a9ad9c6d75e2e6)

**Review by vreynolds — APPROVED** [review]
she-ready

**Pull request merged**